core/src/action.cpp

```diff
@@ -0,0 +1,140 @@
+/*
+  Copyright:    © 2023 SIL International.
+  Description:  Implementation of the action API functions using internal
+                data structures and functions.
+  Create Date:  23 Oct 2023
+  Authors:      Marc Durdin (MCD)
+  History:      23 Oct 2023 - MCD - Initial implementation from #9720
+*/
+#include <cassert>
+#include <algorithm>
+#include <sstream>
+
+#include <keyman/keyman_core_api.h>
+
+#include "action.hpp"
+#include "state.hpp"
+#include "option.hpp"
+
+km_core_actions * km::kbp::action_items_to_actions(
+  km_core_action_item const *action_items
+) {
+  assert(action_items != nullptr);
+  if(action_items == nullptr) {
+    return nullptr;
+  }
+
+  km_core_status status = KM_CORE_STATUS_OK;
+
+  km_core_actions* actions = new km_core_actions;
```

Comment:
```suggestion
  std::unique_ptr<km_core_actions> actions(new km_core_actions);
```

Reply:
Hmm, how do we return this object then? std::move?

Reply:
Thanks @ermshiperete ... guess I am having a bad C++ day 😆